File: api/Enkl.Api/Services/EnterpriseSettingsSerializer.cs
using System.Text.Json;
using Enkl.Api.Dtos;

namespace Enkl.Api.Services;

/// <summary>
/// Shared camelCase (de)serialization for Organisation.EnterpriseSettingsJson — the small subset of
/// App Settings' "Enterprise" category that applies org-WIDE (Forms & Workflow, Portfolio Planner,
/// Portals) rather than per-project like every other App Settings toggle. Same shape/convention as
/// ProjectSettingsSerializer, just a different (organisation-scoped) column and a much smaller field
/// set. Every field is opt-in (default false) — a missing/corrupted value must never silently turn
/// on a module no Org Admin has ever actually switched on.
/// </summary>
public static class EnterpriseSettingsSerializer
{
    private static readonly JsonSerializerOptions Options = new(JsonSerializerDefaults.Web);

    public static string Serialize(EnterpriseSettingsDto settings) => JsonSerializer.Serialize(settings, Options);

    public static EnterpriseSettingsDto Parse(string? json)
    {
        JsonDocument? doc = null;
        try
        {
            if (!string.IsNullOrWhiteSpace(json)) doc = JsonDocument.Parse(json);
        }
        catch (JsonException)
        {
            // Corrupted/garbled JSON falls through to defaults below, same as the client-side guard.
        }

        bool Get(string name)
        {
            if (doc is not null && doc.RootElement.TryGetProperty(name, out var val) &&
                (val.ValueKind == JsonValueKind.True || val.ValueKind == JsonValueKind.False))
            {
                return val.ValueKind == JsonValueKind.True;
            }
            return false;
        }

        var result = new EnterpriseSettingsDto(Get("forms"), Get("portfolioPlanner"), Get("portals"), Get("resources"));
        doc?.Dispose();
        return result;
    }
}

File: php-api/src/Services/EnterpriseSettingsSerializer.php
<?php

declare(strict_types=1);

namespace Enkl\Api\Services;

/**
 * Ported from Services/EnterpriseSettingsSerializer.cs. Shared camelCase (de)serialization for
 * Organisations.EnterpriseSettingsJson — the small subset of App Settings' "Enterprise" category
 * that applies org-WIDE (Forms & Workflow, Portfolio Planner, Portals) rather than per-project like
 * every other App Settings toggle. Every field is opt-in (default false) — a missing/corrupted value
 * must never silently turn on a module no Org Admin has ever actually switched on.
 *
 * @phpstan-type EnterpriseSettings array{forms:bool,portfolioPlanner:bool,portals:bool,resources:bool}
 */
final class EnterpriseSettingsSerializer
{
    private const DEFAULTS = [
        'forms' => false,
        'portfolioPlanner' => false,
        'portals' => false,
        'resources' => false,
    ];

    public static function serialize(array $settings): string
    {
        $result = [];
        foreach (self::DEFAULTS as $key => $default) {
            $result[$key] = (bool) ($settings[$key] ?? $default);
        }
        return json_encode($result);
    }

    /** @return EnterpriseSettings */
    public static function parse(?string $json): array
    {
        $decoded = [];
        if ($json !== null && $json !== '') {
            $tmp = json_decode($json, true);
            if (is_array($tmp)) {
                $decoded = $tmp;
            }
        }

        $result = [];
        foreach (self::DEFAULTS as $key => $default) {
            $value = $decoded[$key] ?? null;
            $result[$key] = is_bool($value) ? $value : $default;
        }
        return $result;
    }
}
